fileServer.js

```diff
@@ -10,31 +10,33 @@ const mimeType = {
   'jpg': 'image/jpeg',
   'png': 'image/png',
   'js': 'text/javascript',
-  'css': 'text/css'
+  'css': 'text/css',
+  'svg': 'image/svg+xml',
+  'ico': 'image/x-icon',
+  'gif': 'image/gif',
 };
 
-function generate500error(response, errorMessage) {
-
-  Error.captureStackTrace(this, generate500error);
+function generate500error(response) {
   var err = new Error();
-  response.writeHead(500, {
-    'Content-Type': 'text/html'
-  });
+  fs.readFile('public/500.html', 'UTF-8', function(error, html) {
+    if (error) {
+      throw err;
```

Comment:
Much cleaner logic :) - however, we are in a critical error handling routine - throwing an error means the server dies and there is no logging - that will not be friendly for the poor operations people who will be waking up at 2 AM when this happens. Come up with something that still exits early, but logs the error to the console *and* returns the error to the user with a 500 code (since we are in an internal server error handler already).